compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java

```diff
@@ -257,7 +262,8 @@ static String getStrandMetadataVarName(String parentFunction) {
     }
 
     public static String cleanupFunctionName(String functionName) {
-        return functionName.replaceAll("[\\.:/<>]", "_");
+        return functionName.matches("(.*)[\\.:/<>](.*)") ? "$" + functionName.replaceAll("[\\.:/<>]", "_") :
```

Comment:
add todo saying remove after adding $ to all generated names in front-end
create an issue and link it